package.json

```
@@ -10,7 +10,9 @@
   "scripts": {
     "runner": "npx ts-node -r tsconfig-paths/register",
     "contentlayer:build": "npx contentlayer2 build --clearCache",
-    "dev": "yarn contentlayer:build && next dev -p 3001",
+    "code-import": "npx esrun coder.ts",
+    "dev": "concurrently -p \"[{name}]\" -n \"code import,Contentlayer,next dev\" -c \"bgBlue.bold,bgMagenta.bold,bgGreen.bold\" \"yarn code-import --watch\" \"yarn contentlayer:build\" \"next dev -p 3001\"",
```

there is a race condition in the concurrently due to content layer not generating fast enough before the nextjs site is ready, contentlayer is required to complete before nextjs should run to avoid the errors

the code import command will not need to run everytime the dev command will be run. so lets keep the original dev command and add a new one that runs the import concurrently:

```suggestion
    "dev": "yarn contentlayer:build && next dev -p 3001",
    "dev:importer": "concurrently -p \"[{name}]\" -n \"code import,next dev\" -c \"bgBlue.bold,bgGreen.bold\" \"yarn code-import --watch\" \"yarn dev\"",
```

---

what is the watch flag supposed to do? because it does not seem to do anything for me

---

Watch flags update the code whenever any markdown file is changed.

---

I want to suggest a better way to do this, 
we can do build first and then use concurrently with crowdin excluded

---

I still think my suggested changes here are the best experience here. it solves all the DX issues which are still here with your current committed changes